core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -967,7 +967,7 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
       // ZK-based
       if (migrationEnabled) {
         require(brokerId >= 0,
-          "broker broker.id.generation.enable is incompatible with migration to ZK. Please disable it before enabling migration")
+          "broker.id generation is incompatible with ZooKeeper migration. Please stop using it before enabling migration (set broker.id to a value greater or equal to 0).")
```

Comment:
this path is unrelated to broker.id generation, right?

Reply:
I'm not entirely sure why it is here -> I just updated the message. The test does not trigger it. But to be honest, my Kafka code knowledge is not good enough to understand if under some circumstances it might get here.

Reply:
This path and the above one (L850) are both under `validateValues` method. The above one is validate for ZK brokers (i.e. process.role is null). And this line is to validate KRaft, or migrating brokers. Both are related to this fix.

Reply:
Maybe we should just say "please define the broker.id for migration"? It can fail due to undefined broker.if even though we have disabled generation.

Reply:
You mean in the error message? Something like `Please define the broker.id for ZooKeeper migration`?

Reply:
never mind. Current version is good enough. I will merge it later